vhdbuilder/scripts/automate_generate_version.sh

```diff
@@ -4,22 +4,39 @@ set -uexo pipefail
 VHD_BUILD_ID="${VHD_BUILD_ID:-""}"
 
 get_image_version_from_publishing_info() {
-    # Note that we previously would specify potentially more than one build, rather than just VHD_BUILD_ID
-    # when doing so, we made sure that all publishing-info's coming from the different builds had the same image version.
-    # this was needed when we previously released VHDs as a single batch from multiple builds.
-    # if we ever need to do that again we should add back that validation.
+    local unique_image_version
+
     for artifact in $(az pipelines runs artifact list --run-id $VHD_BUILD_ID | jq -r '.[].name'); do # Retrieve what artifacts were published
         if [[ $artifact == *"publishing-info"* ]]; then
-            # just take the image version from the first publishing-info we find (since they should all be the same)
-            # TODO(cameissner): add image version validation to separate validation template
-            az pipelines runs artifact download --artifact-name $artifact --path $(pwd) --run-id $VHD_BUILD_ID
-            GENERATED_IMAGE_VERSION=$(jq -r .image_version < vhd-publishing-info.json)
+            # remove publishing info downloaded from some previous iteration, if it exists
             rm -f vhd-publishing-info.json
-            return 0
+
+            az pipelines runs artifact download --artifact-name $artifact --path $(pwd) --run-id $VHD_BUILD_ID
+            image_version=$(jq -r image_version < vhd-publishing-info.json)
+            if [ -z "$unique_image_version" ]; then
+                unique_image_version=$image_version
+                continue
+            fi
+
+            if [ "$image_version" != "$unique_image_version" ]; then
+                # this is to ensure that all publishing-infos coming from the same VHD build specify the same image_version,
+                # mismatching image_versions will cause problems downstream in the release process
+                echo "mismatched image version for VHD build with ID: $VHD_BUILD_ID"
+                echo "expected publishing info artifact $artifact to have image_version $unique_image_version, but had: $image_version"
```

Comment:
what do we need to do when this occur / should we add a simple one liner instruction ?

Reply:
unfortunately they'd need to run a new build entirely from scratch :(

Reply:
added an instruction